pkg/yurtctl/cmd/yurtinit/init.go

```diff
@@ -19,620 +18,286 @@ package yurtinit
-			svcSubnetCIDR, err := kubeadmconstants.GetKubernetesServiceCIDR(d.cfg.Networking.ServiceSubnet, features.Enabled(d.cfg.FeatureGates, features.IPv6DualStack))
-			if err != nil {
-				return nil, errors.Wrapf(err, "unable to get internal Kubernetes Service IP from the given service CIDR (%s)", d.cfg.Networking.ServiceSubnet)
-			}
-			// If we're dry-running, we should create a faked client that answers some GETs in order to be able to do the full init flow and just logs the rest of requests
-			dryRunGetter := apiclient.NewInitDryRunGetter(d.cfg.NodeRegistration.Name, svcSubnetCIDR.String())
-			d.client = apiclient.NewDryRunClient(dryRunGetter, os.Stdout)
-		} else {
-			// If we're acting for real, we should create a connection to the API server and wait for it to come up
-			var err error
-			d.client, err = kubeconfigutil.ClientSetFromFile(d.KubeConfigPath())
-			if err != nil {
-				return nil, err
-			}
+		comp := "sealer"
+		target := fmt.Sprintf("/usr/bin/%s", comp)
+		if err := edgenode.CopyFile(TmpDownloadDir+"/"+comp, target, 0755); err != nil {
+			return err
 		}
 	}
-	return d.client, nil
+	return nil
 }
 
-// Tokens returns an array of token strings.
-func (d *initData) Tokens() []string {
-	tokens := []string{}
-	for _, bt := range d.cfg.BootstrapTokens {
-		tokens = append(tokens, bt.Token.String())
-	}
-	return tokens
+// PullClusterImage will try to pull cluster image to lacal.
+// The cluster image will be used later.
+func (ci *clusterInitializer) PullClusterImage() error {
+	clusterImage := fmt.Sprintf(PullInitClusterImage, ci.ImageRepository, ci.OpenYurtVersion)
+	runCmd := fmt.Sprintf(SealerPullCmd, clusterImage)
+	cmd := exec.Command("bash", "-c", runCmd)
+	return execSealerCmd(cmd)
 }
 
-// KustomizeDir returns the folder where kustomize patches for static pod manifest are stored
-func (d *initData) KustomizeDir() string {
-	return d.kustomizeDir
+// InstallCluster initialize the master of openyurt cluster by calling sealer
+func (ci *clusterInitializer) InstallCluster() error {
+	klog.Infof("init an openyurt cluster")
+	runCmd := fmt.Sprintf(SealerRunCmd, TmpDownloadDir)
+	cmd := exec.Command("bash", "-c", runCmd)
+	return execSealerCmd(cmd)
 }
 
-func printJoinCommand(out io.Writer, adminKubeConfigPath, token string, i *initData) error {
-	joinEdgeNodeCommand, err := getJoinCommand(adminKubeConfigPath, token, "edge-node")
+// PrepareClusterfile fill the template and write the Clusterfile to the /tmp
+func (ci *clusterInitializer) PrepareClusterfile() error {
+	klog.Infof("generate Clusterfile for openyurt")
+	err := os.MkdirAll(TmpDownloadDir, dirMode)
 	if err != nil {
 		return err
 	}
-	joinCloudNodeCommand, err := getJoinCommand(adminKubeConfigPath, token, "cloud-node")
+
+	clusterfile, err := tmplutil.SubsituteTemplate(OpenYurtClusterfile, map[string]string{
+		"apiserver_address":         ci.AdvertiseAddress,
+		"cluster_image":             fmt.Sprintf(LocalInitClusterImage, ci.OpenYurtVersion),
```

Comment:
maybe we need to use `PullInitClusterImage ` instead `LocalInitClusterImage `

Reply:
down